From 0c8ea12a5fd2bc0826c41cefc2532a0a7ed20ff5 Mon Sep 17 00:00:00 2001
From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Thu, 31 Jul 2014 12:32:09 +0300
Subject: [PATCH] ASoC: fsl_asrc: fix an error code in fsl_asrc_probe()

There is a cut and paste bug so it returns success instead of the error
code.

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Mark Brown <broonie@linaro.org>
(cherry picked from commit d387dd08e444b22f844475780fe12a1ad1c6fffd)
---
 sound/soc/fsl/fsl_asrc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/fsl/fsl_asrc.c b/sound/soc/fsl/fsl_asrc.c
index 910525b..424b23c 100644
--- a/sound/soc/fsl/fsl_asrc.c
+++ b/sound/soc/fsl/fsl_asrc.c
@@ -826,7 +826,7 @@ static int fsl_asrc_probe(struct platform_device *pdev)
 	asrc_priv->mem_clk = devm_clk_get(&pdev->dev, "mem");
 	if (IS_ERR(asrc_priv->mem_clk)) {
 		dev_err(&pdev->dev, "failed to get mem clock\n");
-		return PTR_ERR(asrc_priv->ipg_clk);
+		return PTR_ERR(asrc_priv->mem_clk);
 	}
 
 	asrc_priv->ipg_clk = devm_clk_get(&pdev->dev, "ipg");
-- 
1.8.0

